snapcraft/plugins/nodejs.py

```diff
@@ -0,0 +1,102 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2015 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The nodejs plugin is useful for node/npm based parts.
+
+The plugin uses node to install dependencies from `package.json`. It
+also sets up binaries defined in `package.json` into the `PATH`.
+
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+
+Additionally, this plugin uses the following plugin specific keywords:
+
+    - node-packages:
+      (list)
+      A list of dependencies to fetch using npm.
+"""
+
+import logging
+import os
+import platform
+
+import snapcraft
+from snapcraft import sources
+
+logger = logging.getLogger(__name__)
+
+_NODEJS_BASE = 'node-v{version}-linux-{arch}'
+_NODEJS_VERSION = '4.2.2'
```

Comment:
Yeah, was my plan to do this in the future, as in different PR, this is already too big as it is. Good thing is 4.2.2 is an LTS